STModCellular.cpp

```diff
@@ -18,53 +18,104 @@
 #include "STModCellular.h"
 #include "mbed_wait_api.h"
 #include "mbed_trace.h"
+#include "DigitalIn.h"
 
 #define TRACE_GROUP "CELL"
 
 using namespace mbed;
 
-STModCellular::STModCellular(FileHandle *fh) : QUECTEL_UG96(fh),
+STModCellular::STModCellular(FileHandle *fh) : STMOD_CELLULAR_MODEM(fh),
     m_powerkey(MBED_CONF_STMOD_CELLULAR_POWER),
     m_reset(MBED_CONF_STMOD_CELLULAR_RESET),
     m_simsel0(MBED_CONF_STMOD_CELLULAR_SIMSEL0),
     m_simsel1(MBED_CONF_STMOD_CELLULAR_SIMSEL1),
     m_mdmdtr(MBED_CONF_STMOD_CELLULAR_MDMDTR)
 {
-    // start with modem disabled
-    m_mdmdtr.write(0);
-    m_simsel0.write(MBED_CONF_STMOD_CELLULAR_SIM_SELECTION);
-    m_simsel1.write(0);
 
+    tr_debug("STModCellular creation\r\n");
+
+    /* Ensure PIN SIMs are set as input */
+    DigitalIn sim_reset(MBED_CONF_STMOD_CELLULAR_SIM_RESET);
```

Comment:
I don't think the state is guaranteed once the object gets destroyed.
These shall probably be members.

Reply:
good point ! will fix